File: src/main/java/com/indiepost/service/ProfileService.java
package com.indiepost.service;

import com.indiepost.dto.ProfileDto;

import java.util.List;

public interface ProfileService {
    ProfileDto findById(Long id);

    List<ProfileDto> findAll();

    List<ProfileDto> findByIds(List<Long> ids);

    void delete(ProfileDto dto);

    void deleteById(Long id);

    void update(ProfileDto dto);
}

File: src/main/kotlin/com/indiepost/service/ProfileService.kt
package com.indiepost.service

import com.indiepost.dto.ProfileDto
import com.indiepost.dto.ProfileSummaryDto
import com.indiepost.enums.Types
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface ProfileService {
    fun findOne(id: Long): ProfileDto?

    fun findByIdIn(ids: List<Long>): List<ProfileDto>

    fun findOneBySlug(slug: String): ProfileDto?

    fun save(dto: ProfileDto): Long?

    fun deleteById(id: Long): Long?

    fun count(type: Types.ProfileType): Int

    fun count(): Int

    fun find(type: Types.ProfileType, pageable: Pageable): Page<ProfileDto>

    fun find(pageable: Pageable): Page<ProfileDto>

    fun getSummaryDtoList(): List<ProfileSummaryDto>
}
